packages/contracts/contracts/escrow/CollateralEscrowV1.sol

```diff
@@ -0,0 +1,245 @@
+        require(_amount > 0, "Deposit amount cannot be zero");
+        _depositCollateral(
+            _collateralType,
+            _collateralAddress,
+            _amount,
+            _tokenId
+        );
+        Collateral storage collateral = collateralBalances[_collateralAddress];
+        collateral._collateralType = _collateralType;
+        collateral._amount = _amount;
+        collateral._tokenId = _tokenId;
+        emit CollateralDeposited(_collateralAddress, _amount);
+    }
+
+    /**
+     * @notice Withdraws a collateral asset from the escrow.
+     * @param _collateralAddress The address of the collateral contract.
+     * @param _amount The amount to withdraw.
+     * @param _recipient The address to send the assets to.
+     */
+    function withdraw(
+        address _collateralAddress,
+        uint256 _amount,
+        address _recipient
+    ) external onlyOwner {
+        require(_amount > 0, "Withdraw amount cannot be zero");
+        Collateral storage collateral = collateralBalances[_collateralAddress];
+        require(collateral._amount > 0, "No collateral balance for asset");
+        _withdrawCollateral(
+            collateral,
+            _collateralAddress,
+            _amount,
+            _recipient
+        );
+        collateral._amount -= _amount;
+        emit CollateralWithdrawn(_collateralAddress, _amount, _recipient);
+    }
+
+    function _depositCollateral(
+        CollateralType _collateralType,
+        address _collateralAddress,
+        uint256 _amount,
+        uint256 _tokenId
+    ) internal {
+        // Deposit ERC20
+        if (_collateralType == CollateralType.ERC20) {
+            SafeERC20Upgradeable.safeTransferFrom(
+                IERC20Upgradeable(_collateralAddress),
+                msg.sender,
+                address(this),
+                _amount
+            );
+            return;
+        }
+        // Deposit ERC721
+        if (_collateralType == CollateralType.ERC721) {
+            require(_amount == 1, "Incorrect deposit amount");
+            IERC721Upgradeable(_collateralAddress).safeTransferFrom(
+                msg.sender,
+                address(this),
+                _tokenId
+            );
+            return;
+        }
+        // Deposit ERC1155
+        if (_collateralType == CollateralType.ERC1155) {
+            bytes memory data;
+
+            IERC1155Upgradeable(_collateralAddress).safeTransferFrom(
+                msg.sender,
+                address(this),
+                _tokenId,
+                _amount,
+                data
+            );
+            return;
+        }
+    }
+
+    function _withdrawCollateral(
+        Collateral memory _collateral,
+        address _collateralAddress,
+        uint256 _amount,
+        address _recipient
+    ) internal {
+        // Withdraw ERC20
+        if (_collateral._collateralType == CollateralType.ERC20) {
+            SafeERC20Upgradeable.safeTransferFrom(
```

Comment:
can make this normal `transfer` function since this contract is the current owner